components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.publisher/src/main/java/org/wso2/carbon/apimgt/usage/publisher/RequestDataPublisher.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package org.wso2.carbon.apimgt.usage.publisher;
+
+import org.wso2.am.analytics.publisher.reporter.CounterMetric;
+import org.wso2.carbon.apimgt.usage.publisher.dto.AnalyticsEvent;
+
+/**
+ * interface for data publisher
+ */
+public interface RequestDataPublisher {
+
+    void publish(AnalyticsEvent analyticsEvent);
+
+    CounterMetric getCounterMetric();
```

Comment:
Do we really need this method in the interface? This gets implemented by the respective Publishers in a Publisher specific manner. Why can't we bring this to the constructor?

Reply:
I thought it was easy to manage, When we have multiple types of events, we can use any customized CounterMetric. But without any additional changes, abstract class can publish that data.

Reply:
If we don't have a scenario where we have to change the getCounterMetric method only (while keeping the rest of the code the same) better remove this method.